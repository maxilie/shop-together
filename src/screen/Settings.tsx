import SettingsHeader from '../component/header/SettingsHeader';
import SubmitBtn from '../component/button/SubmitBtn';
import { SizeConstants } from '../constant/SizeConstants';
import React from 'react';
import { StyleSheet, View } from 'react-native';
import { ScreenSizeProp, withScreenSize } from '../context/ScreenSizeContext';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { THEME_COLORS } from '../constant/ThemeConstants';

type Props = {
  signOutFunc: () => void;
}

const Settings = (props: Props & ScreenSizeProp) => {

  // Get insets
  const insets = useSafeAreaInsets();

  // Calculate component dimensions
  const screenWidth = props.screenSize.screenWidth;
  const screenHeight = props.screenSize.screenHeight;
  const mainHeaderContentHeight = screenHeight * SizeConstants.FEED.headerHeight;
  const settingsHeaderContentHeight = mainHeaderContentHeight * SizeConstants.FEED.headerHeight;
  const submitBtnHeight = screenHeight * SizeConstants.SIGN_IN_OUT.tabAreaHeight * SizeConstants.SIGN_IN_OUT.fieldHeight * SizeConstants.SIGN_IN_OUT.submitBtnHeight;

  // Define dynamic styles
  const dynStyles = StyleSheet.create({
    contentContainer: {
      flex: 1,
      paddingTop: settingsHeaderContentHeight + insets.top,
      backgroundColor: 'white',
      justifyContent: 'center',
      alignItems: 'center',
    },
  });

  // Render the component...
  return (
    <>

      {/**
       * Content container
       */}
      <View
        style={dynStyles.contentContainer}>

        {/**
         * Sign out button
         */}
        <SubmitBtn
          onSubmit={props.signOutFunc}
          width={screenWidth * SizeConstants.SIGN_IN_OUT.submitBtnWidth}
          height={submitBtnHeight}
          bntText='Sign Out'
          marginTop={0} />
      </View>

      {/**
       * Header
       */}
      <View
        style={{ position: 'absolute', }}>
        <SettingsHeader signOutFunc={props.signOutFunc} />
      </View>

    </>
  );
};

export default withScreenSize(Settings);
