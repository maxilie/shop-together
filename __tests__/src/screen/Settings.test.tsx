import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import { RendererProvider } from 'react-fela';
import { createRenderer } from 'fela-native';
import Settings from '../../../src/screen/Settings';

describe('Settings component', () => {
  it('renders correctly', () => {
    const felaRenderer = createRenderer();
    const tree = renderer.create(
      <RendererProvider renderer={felaRenderer}>
        <NavigationContainer>
          <Settings />
        </NavigationContainer>
      </RendererProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
