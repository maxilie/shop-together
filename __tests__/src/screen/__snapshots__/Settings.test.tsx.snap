// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
        "paddingTop": 13.9932,
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": AnimatedValue {
                " __value": 1,
              },
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4585c4",
            "borderRadius": 10.99791,
            "height": 54.989549999999994,
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 0,
            "shadowColor": "#4585c4",
            "shadowOffset": Object {
              "height": 4.399164,
              "width": 0,
            },
            "shadowOpacity": 0.17,
            "shadowRadius": 16.496864999999996,
            "width": 1433.6000000000001,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 27.494774999999997,
              "fontWeight": "bold",
            }
          }
        >
          Sign Out
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fafbfb",
          "borderBottomColor": "#fafaff",
          "borderBottomWidth": 1.0764,
          "height": 53.82,
          "shadowColor": "lightgray",
          "shadowOffset": Object {
            "height": 5.382000000000001,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 13.9932,
          "width": 1792,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
            "height": 53.82,
            "marginTop": 0,
            "width": 1792,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#363636",
              "fontSize": 27.9864,
              "fontWeight": "bold",
              "marginLeft": "auto",
              "marginRight": "auto",
            }
          }
        >
          Settings
        </Text>
      </View>
    </View>
  </View>,
]
`;
